pkg/model/config/config.go

```diff
@@ -125,11 +127,45 @@ func (c Config) HasGroup(group string) bool {
 	return false
 }
 
+// CRDVersionCompatible returns true if crdVersion can be added to the existing set of CRD versions.
+func (c Config) CRDVersionCompatible(crdVersion string) bool {
+	return c.resourceAPIVersionCompatible("crd", crdVersion)
+}
+
+// WebhookVersionCompatible returns true if webhookVersion can be added to the existing set of Webhook versions.
+func (c Config) WebhookVersionCompatible(webhookVersion string) bool {
+	return c.resourceAPIVersionCompatible("webhook", webhookVersion)
+}
+
+// resourceAPIVersionCompatible returns true if version can be added to the existing set of versions
+// for a given verType.
+func (c Config) resourceAPIVersionCompatible(verType, version string) bool {
+	if len(c.Resources) == 0 {
+		return true
+	}
+	for _, res := range c.Resources {
+		var currVersion string
+		switch verType {
+		case "crd":
+			currVersion = res.CRDVersion
+		case "webhook":
+			currVersion = res.WebhookVersion
+		}
+		if currVersion != "" && version != currVersion {
+			return false
+		}
+	}
+	return true
+}
+
 // GVK contains information about scaffolded resources
 type GVK struct {
 	Group   string `json:"group,omitempty"`
 	Version string `json:"version,omitempty"`
 	Kind    string `json:"kind,omitempty"`
+
+	CRDVersion     string `json:"crdVersion,omitempty"`
+	WebhookVersion string `json:"webhookVersion,omitempty"`
```

Comment:
Why the webhook apiVersion is stored with the resource? 
The crd is a kind of object and webhooks is another. Could we not have more than 1 webhook for the same resource? 
See: https://github.com/kubernetes-sigs/kubebuilder/pull/1696#issuecomment-712143182

Reply:
We could have more than one webhook per resource, and they would all have to have the same version, so this is fine IMO.

Reply:
How we could we track more than 1 webhook for the same resource if the webhookVersion is an attribute of the resource and no the of the webhook? This model does not allow us to work on with diff API versions for webhooks and resources in the future or have N webhooks for the same resource.

Reply:
I cacher your idea.